### DISCUSSION TITLE
Externally-usable theme function

### DISCUSSION THREAD

**Discussion by leo-petrucci** [tailwindlabs/tailwindcss#5742] [Ideas]
I'm unsure if this is already part of Tailwind, but I couldn't find any info about it.

So, in `tailwind.config` we have access to the `theme()` function, to which we can pass any dot notation configuration object and it will return its value making it easy to avoid repetition and for example defining color values in more places.

While working with Tailwind I've found myself in the situation where I want to use one of my defined Tailwind colors outside of a class. For example, I'm building a component with Radix UI and it requires I assign styles to it with `&[data-state="checked"]`.

```
const StyledSwitch = styled(SwitchPrimitive.Root, {
  all: 'unset',
  width: 42,
  height: 25,
  backgroundColor: blackA.blackA9,
  borderRadius: '9999px',
  position: 'relative',
  boxShadow: `0 2px 10px ${blackA.blackA7}`,
  WebkitTapHighlightColor: 'rgba(0, 0, 0, 0)',
  '&:focus': { boxShadow: `0 0 0 2px black` },
  '&[data-state="checked"]': { backgroundColor: 'black' }, // <-- right here
});
```

If I wanted to theme this accordingly to my site, I would need to copy the value assigned to my primary color and copy it in whatever css-in-js solution I'm using which would then make it a bit difficult to maintain.

Of course the easy solution to this is to have a separate config where all the brand colors are kept, but I think it would be neat if Tailwind was the single source of truth for all the colors.

Is this something that could be useful or is it entirely out of scope?

**Comment by xanderificnl** [comment]
Could you handle this with CSS variables? i.e.

`'&[data-state="checked"]': { backgroundColor: '@var(--bg-black)' },`

If so, check out the docs here: https://tailwindcss.com/docs/customizing-colors#using-css-variables
